### DISCUSSION TITLE
___call_main is missing

### DISCUSSION THREAD

**Discussion by k1nder10** [emscripten-core/emscripten#17144] [Q&A]
Look at the following code:

main.cpp
#include <iostream>
int main() {
  std::cout << "Hello, emscripten!\n";
}

emcc main.cpp -o main.html -s MAIN_MODULE=1 -s USE_PTHREADS=1
When building it I'm getting an error "warning: ___call_main may need to be added to EXPORTED_FUNCTIONS if it arrives from a system library". It comes up only if I combine main_module + pthread flags. I tried to add __call_main/_main to EXPORTED_FUNCTION but it didn't help. Is it expected behavior? What am I missing here?

emcc --version
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 3.1.8 (3ff7eff373d31d8c0179895165462019221f192e)

**Comment by sbc100** [comment]
`__call_main` is called by `system/lib/pthread/emscripten_proxy_main.c` when proxy main to another thread.  

I think the problem is that ` -sMAIN_MODULE=1` is force including all the C library code including `emscripten_proxy_main.c` even though PROXY_TO_PTHREAD is not used.

Can you try with `-sMAIN_MODULE=2`.. this will produce a much smaller output and is highly recommended for this reason.

**Reply by k1nder10** [reply]
Yes, MAIN_MODULE=2 helps here. Suppressing this error helps as well. 
Though it's strange that one 'cannot' use MAIN_MODULE=1 & pthreads together...

**Reply by sbc100** [reply]
This bug was fixed in https://github.com/emscripten-core/emscripten/pull/17153
